hash.c

```diff
@@ -2814,14 +2823,24 @@ rb_hash_slice(int argc, VALUE *argv, VALUE hash)
 }
 
 /*
- * call-seq:
- *   hsh.values_at(key, ...)   -> array
+ *  call-seq:
+ *    hash.values_at(*keys) -> new_array
+ *
+ *  Returns a new \Array containing values for the given +keys+:
+ *    h = {foo: 0, bar: 1, baz: 2}
+ *    h.values_at(:foo, :baz) # => [0, 2]
  *
- * Return an array containing the values associated with the given keys.
- * Also see Hash.select.
+ *  Returns an empty Array if no arguments given:
+ *    h = {foo: 0, bar: 1, baz: 2}
+ *    h.values_at # => []
+ *
+ *  ---
  *
- *   h = { "cat" => "feline", "dog" => "canine", "cow" => "bovine" }
- *   h.values_at("cow", "cat")  #=> ["bovine", "feline"]
+ *  Raises an exception if any given key is invalid
+ *  (see {Invalid Hash Keys}[#class-Hash-label-Invalid+Hash+Keys]):
+ *    h = {foo: 0, bar: 1, baz: 2}
+ *    # Raises NoMethodError (undefined method `hash' for #<BasicObject:>)
```

Comment:
Is the colon in the right place?

Reply:
```
irb(main):001:0> h = {foo: 0, bar: 1, baz: 2}
irb(main):002:0> h.values_at(BasicObject.new)
Traceback (most recent call last):
        5: from /home/samuel/.rubies/ruby-2.7.1/bin/irb:23:in `<main>'
        4: from /home/samuel/.rubies/ruby-2.7.1/bin/irb:23:in `load'
        3: from /home/samuel/.rubies/ruby-2.7.1/lib/ruby/gems/2.7.0/gems/irb-1.2.3/exe/irb:11:in `<top (required)>'
        2: from (irb):2
        1: from (irb):2:in `values_at'
NoMethodError (undefined method `hash' for #<BasicObject:0x000055fcc63cdfc0>)
```

For formatting this, I might suggest just writing `#<BasicObject>` because the colon serves no purpose.